internal/host/validator.go

```diff
@@ -1714,3 +1716,54 @@ func (v *validator) printVSphereUUIDEnabled(_ *validationContext, status Validat
 		return fmt.Sprintf("Unexpected status %s", status)
 	}
 }
+
+func (v *validator) serviceCanConnectToKubeAPI(c *validationContext) ValidationStatus {
+	// If we do not have a cluster then we definitely are not in Day2 status!
+	if c.cluster == nil {
+		return ValidationSuccessSuppressOutput
+	}
+
+	var status ValidationStatus
+
+	// Check for the "Day2" status "adding-hosts"
+	addingHosts := models.ClusterStatusAddingHosts
+
+	// If we are not in day2 then this validation does not apply
+	// TODO: Only applies if Kube-API is enabled - we need to add this to the check...
+	if !c.kubeApiEnabled || (c.cluster != nil && c.cluster.Status != &addingHosts) {
+		status = ValidationSuccessSuppressOutput
+	} else {
```

Comment:
This was a deliberate style choice as when the "Validation merge" https://github.com/openshift/assisted-service/pull/4097 PR is on master, this will merge the validation and print methods. At which point returning in this way will make less sense.